alvr/client_core/src/platform/android/decoder.rs

```diff
@@ -193,10 +188,12 @@ fn decoder_attempt_setup(
     Ok(decoder)
 }
 
+// Since we leak the ImageReader, and we pass frame_result_callback to it which contains a reference
+// to ClientCoreContext, to avoid circular references we need to use a Weak reference.
 fn decoder_lifecycle(
     config: DecoderConfig,
     csd_0: Vec<u8>,
-    dequeued_frame_callback: impl Fn(Duration) + Send + 'static,
+    frame_result_callback: Weak<impl Fn(Result<Duration>) + Send + Sync + 'static>,
```

Comment:
Kinda borrowing this from a recent wine review but isn't weak a sign that our locking requirements have gotten outta hand could you justify it?

Reply:
Seems there is actually no way to extract the underlying AImageReader from the ImageReader so we would have to do our custom wrapper